GetIntoTeachingApi/Jobs/ApplyBackfillJob.cs

```diff
@@ -76,7 +88,48 @@ private async Task QueueCandidateSyncJobs(DateTime updatedSince, int startPage)
             // to process the next batch of pages.
             if (paginator.HasNext)
             {
-                _jobClient.Enqueue<ApplyBackfillJob>((x) => x.RunAsync(updatedSince, paginator.Page));
+                _jobClient.Enqueue<ApplyBackfillJob>((x) => x.RunAsync(updatedSince, paginator.Page, null));
+            }
+        }
+        
+        private async Task QueueCandidateSyncJobsCandidateIds(IEnumerable<int> candidateIds)
+        {
+            // Enforce use of the Newtonsoft Json serializer
+            FlurlHttp.Clients.UseNewtonsoft();
+            
+            var batch = candidateIds.Take(RecordsPerJob);
+            var remainder = candidateIds.Skip(RecordsPerJob);
+
+            foreach (int candidateId in batch) 
+            {
+                _logger.LogInformation("Fetching CandidateID C{CandidateId} from the Apply API", candidateId);
+
+                try
+                {
+                    var request = Env.ApplyCandidateApiUrl
+                        .AppendPathSegment("candidates")
+                        .AppendPathSegment(String.Format(CultureInfo.InvariantCulture, "C{0}", candidateId))
+                        .WithOAuthBearerToken(Env.ApplyCandidateApiKey);
+
+                    var candidate = await request.GetJsonAsync<Response<GetIntoTeachingApi.Models.Apply.Candidate>>();
+
+                    _logger.LogInformation("Scheduling ApplyBackfillJob - Syncing CandidateID: C{Id}", candidateId);
+                    
+                    _jobClient.Schedule<ApplyCandidateSyncJob>(x => x.Run(candidate.Data), TimeSpan.FromSeconds(60));
+                }
+                catch (FlurlHttpException ex)
+                {
+                    _logger.LogError("Failed to fetch CandidateID C{CandidateId} from the Apply API (status: {Status})", candidateId, ex.StatusCode);
+                }
+
+                await Task.Delay(100);
+            }
+            
+            // When we reach the end page we re-queue the backfill job
+            // to process the next batch of candidate IDs.
+            if (remainder?.Any() ?? false)
```

Comment:
```suggestion
            if (remainder?.Any())
```

Reply:
i'm guessing `null` if falsey in C#

Reply:
Sadly not, a fun fact about C# - `null` does not cast to a boolean:
```
public static void Main()
{
	if (null) {
		Console.WriteLine("Hello World");
	}
}
```

is a compilation error: `Compilation error (line 7, col 7): Cannot convert null to 'bool' because it is a non-nullable value type`


instead, you can use the double question mark operator `??` to return false if the previous arg was a null, e.g.

```
some_value_which_might_be_null ?? false
```

...of course, Ruby solved this problem 30 years ago...

Reply:
dear lord...